pkg/autoscaling/handler.go

```
@@ -50,12 +52,26 @@ func (h *HTTPHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	log.Debug(fmt.Sprintf("autoscale: strategy data: %s", string(data)))
+
 	strategy := Strategy{}
 	if err := json.Unmarshal(data, &strategy); err != nil {
 		h.rd.JSON(w, http.StatusBadRequest, err.Error())
 		return
 	}
 
-	plan := calculate(rc, h.svr.GetPDServerConfig(), &strategy)
-	h.rd.JSON(w, http.StatusOK, plan)
+	plans, err := calculate(rc, &strategy)
+	if err != nil {
+		h.rd.JSON(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	data, err = json.Marshal(plans)
+	if err != nil {
+		log.Error("marshal plans failed.", errs.ZapError(err))
```

need to return an error?